core/src/main/java/ai/timefold/solver/core/config/solver/monitoring/SolverMetric.java

```diff
@@ -90,10 +93,11 @@ public enum SolverMetric {
         this.isConstraintMatchBased = isConstraintMatchBased;
     }
 
-    public String getMeterId() {
+    public @NonNull String getMeterId() {
         return meterId;
     }
 
+    @NullMarked
     public static void registerScoreMetrics(SolverMetric metric, Tags tags, ScoreDefinition<?> scoreDefinition,
             Map<Tags, List<AtomicReference<Number>>> tagToScoreLevels, Score<?> score) {
```

Comment:
@triceo schemagen failed when annotating the `List<AtomicReference<Number>>` param, so I used `@NullMarked` here. 

However, this introduced a new IDE warning (not sure if this is a compiler warning too) on line `116`, since `Metrics.gauge()` is annotated to return a nullable type. From reading the code of `gauge`, I understand this shouldn't be the case when passing a non-null reference in as the third param (which `new AtomicReference` certainly is). 

So I guess we could: 
1. leave it as it is
2. wrap `gauge` with `Objects.requireNonNull` 
3. wrap `gauge` with `Objects.requireNonNullOrElse` (but I guess this might be a bit confusing, since the else-branch should never be reached)

Any thoughts?

Reply:
I'd leave it for now. 

I'll have some downtime before the release, I'll go back to my notes and review some of the places where maybe I want to think about it a bit more. This will be one of them.